### PR TITLE
#166816104 Implement User Update profile feature

### PR DESCRIPTION
**Pull request by fxola**
#### What does this PR do?
- Implement User Update profile feature

#### Description of Task to be completed?
- write migration files to add columns for bio,twitterHandle and facebookHandle
- write service method to find user by id
- write service method to find user by username
- write controller logic to handle user profile update
- write middleware file to parse file(image) input
- write middlware file for profile update validation
- write image helper file to handle image upload to cloudinary
- update auth route to accomodate new route
- write unit tests for profile update feature
- delivers[#166816104]

#### How should this be manually tested?
- Clone the repository using ```git clone https://github.com/andela/persephone-ah-backend.git .``` in your terminal
- run ```npm install``` on your terminal
- run ```npm run start``` to start up the server
- run ```npm test``` to see test results
#### Any background context you want to provide?
-N/A
#### What are the relevant pivotal tracker stories?
- PT Story link - [#166816104](https://www.pivotaltracker.com/n/projects/2353116/stories/166816104)

#### Screenshots (if appropriate)
- N/A

#### Questions:
- N/A

### CODE REVIEW

**Comment by dharmykoya** [comment]
Looks good to me

**Review by halimahO — COMMENTED** [review]
LGTM

**Review by devPinheiro — APPROVED** [review]
LGTM

File: src/controllers/auth.controllers.js [hunk 1]
```diff
@@ -1,60 +1,178 @@
...
+        const imageResponse = await upload(imagePath, imageUniqueName);
+        uploadedImage = imageResponse.secure_url;
+      }
+
+      const fields = {
+        ...request.body,
+        image: uploadedImage || previousImage
+      };
+      const updatedUser = await request.foundUser.update(fields);
+      const {
```
Comment (nedemenang) [inline src/controllers/auth.controllers.js]: DRY

File: src/helpers/image.helper.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+  api_key: process.env.CLOUDINARY_API_KEY,
+  api_secret: process.env.CLOUDINARY_API_SECRET
+});
+
+export const upload = async (path, uniqueFilename) => {
+  return cloudinary.v2.uploader.upload(
+    path,
+    { public_id: `avatar/${uniqueFilename}`, tags: `avatar` }, // directory and tags are optional
+    image => {
+      console.log('file uploaded to Cloudinary');
```
Comment (nedemenang) [inline src/helpers/image.helper.js]: remove redundant console.log

File: src/middlewares/imageUpload.middleware.js [hunk 3]
```diff
@@ -0,0 +1,31 @@
+import multer from 'multer';
+
+const fileFilter = (req, file, cb) => {
```
Comment (nedemenang) [inline src/middlewares/imageUpload.middleware.js]: change from `req` to `request`

File: src/routes/v1/index.js [hunk 4]
```diff
@@ -1,5 +1,7 @@
 import auth from './auth.route';
+import users from './user.routes';
 
 export default app => {
   app.use('/api/v1/users', auth);
+  app.use('/api/v1/users/', users);
```
Comment (nedemenang) [inline src/routes/v1/index.js]: abstract out `/api/v1` to a variable and read from there.

File: src/services/auth.service.js [hunk 5]
```diff
@@ -81,15 +94,66 @@ export const loginService = async body => {
...
+ * @method forgetPasswordServices
+ * - Helps generate password reset token and
+     sends mail to the user's email Address containing token and link
+ * @param {object} user - contains required user info to generate password reset token
+ * @returns {promise}
+ */
+
+export const forgotPasswordServices = async user => {
+  const token = await getPasswordResetToken(user);
+  const url = `${process.env.url}/api/v1/users/password_reset?token=${token}`;
```
Comment (nedemenang) [inline src/services/auth.service.js]: abstract `\api\v1` to a variable and read from there.

**Review by nedemenang — CHANGES_REQUESTED** [review]

File: src/services/auth.service.js (hunk 5 above)
Comment (fxola) [reply src/services/auth.service.js]: alright

File: src/middlewares/imageUpload.middleware.js (hunk 3 above)
Comment (fxola) [reply src/middlewares/imageUpload.middleware.js]: alright

File: src/helpers/image.helper.js (hunk 2 above)
Comment (fxola) [reply src/helpers/image.helper.js]: done

File: src/controllers/auth.controllers.js (hunk 1 above)
Comment (fxola) [reply src/controllers/auth.controllers.js]: refactored

**Review by nedemenang — APPROVED** [review]

**Pull request merged**